File: fun/quote.cc.go
{{/*
	This command quotes a given message with optional channel. Usage: `-quote [channel] <message>`.

	Recommended trigger: Command trigger with trigger `quote`
*/}}
{{ $channel := .Channel }}
{{ $id := joinStr " " .CmdArgs }}
{{ if .CmdArgs }}
	{{ $channelID := "" }}
	{{ with reFindAllSubmatches `<#(\d+)>` (index .CmdArgs 0) }} {{ $channelID = index . 0 1 }} {{ end }}
	{{ $temp := getChannel (or $channelID (index .CmdArgs 0)) }}
	{{ if $temp }}
		{{ $channel = $temp }}
		{{ $id = slice .CmdArgs 1 | joinStr " " }}
	{{ end }}
{{ end }}
{{ $id = reFind `^\d+$` $id }}
{{ if $id }}
	{{ $id = toInt64 $id }}
	{{ $msg := getMessage $channel.ID $id }}
	{{ if $msg }}
		{{ if $msg.Content }}
			{{ $img := "" }}
			{{ with $msg.Attachments }} {{ $img = (index . 0).URL }} {{ end }}
			{{ $link := printf "https://discordapp.com/channels/%d/%d/%d" .Guild.ID $channel.ID $msg.ID }}
			{{ sendMessage nil (cembed
				"author" (sdict "name" $msg.Author.String "icon_url" ($msg.Author.AvatarURL "256"))
				"description" (printf "%s\n\n[*Original message*](%s)" $msg.Content $link)
				"color" 14423100
				"footer" (sdict "text" (printf "Requested by %s | Quote from" .User.String))
				"timestamp" $msg.Timestamp
				"image" (sdict "url" $img)
			) }}
		{{ end }}
	{{ else }}
		Please provide a valid message to quote.
	{{ end }}
{{ else }}
	The syntax of this command is `-quote [channel] <message>`.
{{ end }}

File: util/quote.cc.lua
{{/*
Originally published by @Jonas747#0001, regex fixed by (O1)#0001.

This command displays an embed with the contents of a linked discord message.
Usage: Paste a link to a Discord message.

Recommended trigger: Regex trigger with trigger `https?://(?:.+\.)?discord(?:app)?.com/channels\/(\d+)\/(\d+)\/(\d+)`
*/}}

{{ $matches := reFindAllSubmatches `https?://(?:.+\.)?discord(?:app)?.com/channels\/(\d+)\/(\d+)\/(\d+)` .Message.Content }} 
{{$msg := getMessage (index (index $matches 0) 2) (index (index $matches 0) 3) }}
{{if not $msg}}

{{else}}
{{ $avatar := (joinStr "" "https://cdn.discordapp.com/avatars/" (toString $msg.Author.ID) "/" $msg.Author ".png") }}

{{$embedRaw := sdict 
"description" (joinStr "" "**[Message Link](" (index (index $matches 0) 0) ")  to <#" $msg.ChannelID ">**\n" $msg.Content)
"color" 4645612 
"author" (sdict "name"  $msg.Author.Username "icon_url" ($msg.Author.AvatarURL "64")) 
"footer" (sdict "text" (joinStr "" "Req. by "  .Message.Author.Username ". Quote from ")) 
"timestamp" $msg.Timestamp  }}

{{if $msg.Attachments}}
{{$embedRaw.Set "image" (sdict "url" (index $msg.Attachments 0).URL) }}
{{end}}

{{ sendMessage nil (cembed $embedRaw) }}

{{/* delete the trigger if it only contained a link and nothing more */}}
{{if eq (len (index (index $matches 0) 0)) (len .Message.Content) }} {{deleteTrigger 1}} {{end}} 
{{end}}